docs/source/creating_services.rst

```diff
@@ -181,6 +182,63 @@ Dockerfile (optional)
+
+    Service namespace
+        The group to which the service belongs e.g. your name or your organisation's name. If in doubt, use the GitHub
+        handle of the user or organisation publishing the services.
+
+        Namespaces must be lower kebab case (i.e. they may contain the letters [a-z], numbers [0-9], and hyphens [-]).
+        They may not begin or end with hyphens.
+
+    Service name
+        A name to uniquely identify the service within its namespace. This usually corresponds to the name of the GitHub
+        repository for the service. Names must be lower kebab case (i.e. they may contain the letters [a-z],
+        numbers [0-9] and hyphens [-]). They may not begin or end with hyphens.
+
+    Service revision tag
+        A tag that uniquely identifies a particular revision of a service. The revision tag could correspond to a commit
+        hash like ``a3eb45``, a release number like ``0.12.4``, a branch name (e.g. ``development``), a particular
+        environment the service is deployed in (e.g. ``production``), or a combination like ``0.12.4-production``. Tags
+        may contain lowercase and uppercase letters, numbers, underscores, periods, and hyphens, but can't start with a
+        period or a dash. They can contain a maximum of 128 characters. These requirements are the same as the `Docker
+        tag format <https://docs.docker.com/engine/reference/commandline/tag/>`_.
+
+    Service ID
+        The SRUID is a special case of the service ID. A service ID can be an SRUID or just the service namespace and
+        name. It can be used to ask a question to a service without specifying a specific revision of it. This enables
+        asking questions to, for example, the service ``octue/my-service`` and automatically having them routed to its
+        latest revision. Note that this will be a future feature; currently, you will still be required to provide a
+        revision tag (i.e. a full SRUID).
+
+Where to specify the namespace, name, and revision tag
+------------------------------------------------------
+The name and namespace are read from the service's ``octue.yaml`` file. However, to make deployment to particular
+environments easier, they can alternatively be specified via the following environment variables:
+
+- ``OCTUE_SERVICE_NAMESPACE``
+- ``OCTUE_SERVICE_NAME``
+
+These environment variables take precedence over the values in ``octue.yaml``.
+
+Revision tags are specified differently - they're how you keep service revisions unique, so it doesn't make sense to
+hard-code them in ``octue.yaml``. If the revision tag is not specified in the ``OCTUE_SERVICE_REVISION_TAG`` environment
+variable or, if using the ``octue start`` CLI command, via the ``--revision-tag`` CLI option, a "coolname" tag (e.g.
+``hungry-hippo``) will be generated to ensure the uniqueness of the service revision. If the CLI option is provided, it
+takes precedence over the ``OCTUE_SERVICE_REVISION_TAG`` environment variable.
```

Comment:
I'd maybe make this into a table. All the information is there but it's a bit "if this then that" which is dense to read.
```
+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------+
|               | How to set                                                                                                                                                                                                                                                                                                                        | Required                                                                                                                  |
+===============+===================================================================================================================================================================================================================================================================================================================================+===========================================================================================================================+
| Namespace     | Set in ``octue.yaml`, or override that by setting an ``OCTUE_SERVICE_NAMESPACE`` environment variable                                                                                                                                                                                                                             | y                                                                                                                         |
| Name          | Set in ``octue.yaml`, or override that by setting an ``OCTUE_SERVICE_NAME`` environment variable                                                                                                                                                                                                                                  | y                                                                                                                         |
| Revision Tag  | Set an ``OCTUE_SERVICE_REVISION_TAG`` environment variable, or specify using the ``--revision-tag`` flag when starting a service with the CLI (see ``octue start --help`` for more details). You cannot hard-code this in ``octue.yaml`` in order to maintain uniqueness (consider multiple people using the same code at once).  | n - a default "cool-name" tag is generated on service start if no tag is specified, to ensure uniqueness of the service.  |
+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------+
```

Reply:
Just tried this but it renders really poorly because of the insistence of showing all text in one tiny column

Reply:
I'll try something else though